File: app/src/main/java/ua/com/foxminded/locationtrackera/model/locations/dao/UserLocationDao.java
package ua.com.foxminded.locationtrackera.model.locations.dao;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;

import java.util.List;

import ua.com.foxminded.locationtrackera.model.locations.UserLocation;

@Dao
public interface UserLocationDao {

    @Insert
    void saveLocation(UserLocation location);

    @Query("SELECT * FROM UserLocation")
    List<UserLocation> getAllLocations();

    @Query("DELETE FROM UserLocation")
    void deleteAllLocation();
}

File: models_impl/src/main/java/ua/com/foxminded/locationtrackera/models_impl/locations/dao/UserLocationDao.kt
package ua.com.foxminded.locationtrackera.models_impl.locations.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import ua.com.foxminded.locationtrackera.models.locations.UserLocation

@Dao
interface UserLocationDao {
    @Insert
    fun saveLocation(location: UserLocation)

    @Query("SELECT * FROM UserLocation")
    fun getAllLocations(): List<UserLocation>

    @Query("DELETE FROM UserLocation")
    fun deleteAllLocation()
}